src/AutoMapper/Features/Features.cs

```diff
@@ -0,0 +1,40 @@
+﻿using AutoMapper.Configuration;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace AutoMapper.Features
+{
+    public class Features<TFeature> : IEnumerable<TFeature>
+    {
+        private IDictionary<Type, TFeature> _features = new Dictionary<Type, TFeature>();
+
+        /// <summary>
+        /// Gets the feature of type <typeparamref name="TFeatureToFind"/>.
+        /// </summary>
+        /// <typeparam name="TFeatureToFind">The type of the feature.</typeparam>
+        /// <returns>The feature or null if feature not exists.</returns>
+        public TFeatureToFind Get<TFeatureToFind>() where TFeatureToFind : TFeature => (TFeatureToFind)_features.GetOrDefault(typeof(TFeatureToFind));
+
+        /// <summary>
+        /// Add or update the feature. Existing feature of the same type will be replaced.
+        /// </summary>
+        /// <param name="feature">The feature.</param>
+        public void AddOrUpdate(TFeature feature) => _features[feature.GetType()] = feature;
```

Comment:
You're going to hate me - but I think I prefer `Set` instead of `AddOrUpdate`. Then it's `Get`/`Set`.

Reply:
But look at the calling code. Weird :)